actionpack/lib/action_controller/metal/head.rb

```diff
@@ -17,8 +17,14 @@ module Head
     #
     # See Rack::Utils::SYMBOL_TO_STATUS_CODE for a full list of valid +status+ symbols.
     def head(status, options = {})
-      options, status = status, nil if status.is_a?(Hash)
-      status ||= options.delete(:status) || :ok
+      if status.is_a?(Hash)
+        msg = "Passing Hash as first parameter"
+        msg << " and the implicit :ok status" unless status.key?(:status)
+        options, status = status, (status.delete(:status) || :ok)
```

Comment:
As a tiny nitpick I don't like the parens here, I understand you're worried about the execution order. Would it help to follow @gsamokovarov's suggestion and use the block version?